src/main/java/dao/FruitShopDao.java

```diff
@@ -0,0 +1,7 @@
+package dao;
+
+import java.util.List;
+
+public interface FruitShopDao {
+    void processTransactions(List<String> transactionLines);
```

Comment:
dao - data access object. It is for DB, not for processing transactions
```suggestion
public interface FruitShopService {
    void processTransactions(List<FruitTransaction> transactions);
```